posthog/event_usage.py

```diff
@@ -0,0 +1,79 @@
+"""
+Module to centralize event reporting on the server-side.
+"""
+
+import posthoganalytics
+
+from posthog.models import Organization, User
+
+
+def report_user_signed_up(
+    distinct_id: str,
+    is_instance_first_user: bool,
+    is_organization_first_user: bool,
+    new_onboarding_enabled: bool = False,
+    backend_processor: str = "",  # which serializer/view processed the request
+    social_provider: str = "",  # which third-party provider processed the login (empty = no third-party)
+) -> None:
+
+    props = {
+        "is_first_user": is_instance_first_user,
+        "is_organization_first_user": is_organization_first_user,
+        "new_onboarding_enabled": new_onboarding_enabled,
+        "signup_backend_processor": backend_processor,
+        "social_provider": social_provider,
+    }
+
+    # TODO: This should be $set_once as user props.
+    posthoganalytics.identify(distinct_id, props)
+    posthoganalytics.capture(distinct_id, "user signed up", properties=props)
+
+
+def report_onboarding_completed(organization: Organization, current_user: User) -> None:
+    """
+    Reports that the `new-onboarding-2822` has been completed.
+    """
+
+    team_members_count = organization.members.count()
+
+    # TODO: This should be $set_once as user props.
```

Comment:
Q: Why?

Reply:
Well because it doesn't make sense for these properties to change, they are props related to the signup process which should only happen once in a user's lifecycle.